strongbox-cron/strongbox-cron-api/src/main/java/org/carlspring/strongbox/cron/services/impl/CronJobSchedulerServiceImpl.java

```diff
@@ -24,134 +29,97 @@
-
-            jobsMap.put(cronTaskConfiguration.getName(), cronTask);
+            triggerBuilder.withSchedule(CronScheduleBuilder.cronSchedule(cronExpression));
         }
+        Trigger trigger = triggerBuilder.build();
 
-        if (!scheduler.isStarted())
+        try
         {
-            logger.debug("Scheduler started");
+            scheduler.scheduleJob(jobDetail, Collections.singleton(trigger), true);
+        }
+        catch (SchedulerException e)
+        {
+            logger.error(String.format("Failed to add Cron Job:\n [%s]", cronTaskConfiguration), e);
 
-            scheduler.start();
+            return;
         }
 
         logger.debug("Job '" + cronTaskConfiguration.getName() + "' scheduled.");
     }
 
     @Override
     public void executeJob(CronTaskConfigurationDto cronTaskConfiguration)
-            throws SchedulerException
     {
-        Scheduler scheduler = schedulerFactoryBean.getScheduler();
+        JobKey jobKey = JobKey.jobKey(cronTaskConfiguration.getName());
 
-        if (jobsMap.containsKey(cronTaskConfiguration.getName()))
+        try
         {
-            CronTask cronTask = jobsMap.get(cronTaskConfiguration.getName());
-            JobDetail jobDetail = cronTask.getJobDetail();
-
-            scheduler.triggerJob(jobDetail.getKey());
+            scheduler.triggerJob(jobKey);
         }
-        else
+        catch (SchedulerException e)
         {
-            throw new SchedulerException("Could not find cron task with key " + cronTaskConfiguration.getName() + "!");
+            logger.error(String.format("Failed to shcedule cron job [%s]", jobKey));
         }
     }
 
     @Override
     public void deleteJob(String cronTaskConfigurationName)
-            throws ClassNotFoundException, SchedulerException, CronTaskNotFoundException
     {
-        Scheduler scheduler = schedulerFactoryBean.getScheduler();
-        if (!jobsMap.containsKey(cronTaskConfigurationName))
+        TriggerKey triggerKey = TriggerKey.triggerKey(cronTaskConfigurationName);
+        JobKey jobKey = JobKey.jobKey(cronTaskConfigurationName);
+
+        try
         {
-            logger.warn("Could not find cron task under the specified name! Existing jobs map: {}, lookup name: {}", jobsMap,
-                        cronTaskConfigurationName);
-            return;
+            scheduler.unscheduleJob(triggerKey);
+            scheduler.deleteJob(jobKey);
         }
-
-        CronTask cronTask = jobsMap.get(cronTaskConfigurationName);
-        JobDetail jobDetail = cronTask.getJobDetail();
-        Trigger trigger = cronTask.getTrigger();
-
-        scheduler.unscheduleJob(trigger.getKey());
-        scheduler.deleteJob(jobDetail.getKey());
-
-        jobsMap.remove(cronTaskConfigurationName);
-
-        logger.debug("Job '" + cronTaskConfigurationName + "' un-scheduled.");
-    }
-
-    @Override
-    public GroovyScriptNamesDto getGroovyScriptsName()
```

Comment:
So, where is the Groovy support going? What am I missing here?

Reply:
yeah, I have to drop this because of implementation changes
my thoughts was that first we need to make regular CronJob works  and make Groovy support works will be follow up

Reply:
Hm... But the Groovy script suport is supposed to work. I mean -- we tried it and it did invoke Groovy scripts as expected. I'm only raising this because @bamarch is working on testing out the Groovy cron tasks and documenting them. If I understand correctly, the removal of this code will drop the support for Groovy cron tasks. Is this so, or am I misunderstanding something?